Core.ApplicationServices/Users/Write/UserWriteService.cs

```diff
@@ -139,10 +139,47 @@ public Maybe<OperationError> CopyUserRights(Guid organizationUuid, Guid fromUser
                 return fromUser.Error;
             }
 
-            return _userService.GetUserInOrganization(organizationUuid, toUserUuid)
+            using var transactionManager = _transactionManager.Begin();
+            var copyResult = _userService.GetUserInOrganization(organizationUuid, toUserUuid)
                 .Bind(CanModifyUser)
-                .Match(toUser => _userRightsService.CopyRights(fromUser.Value.Id, toUser.Id, org.Value.Id, parameters), 
+                .Match(toUser =>
+                    {
+                        return FilterOutExistingRights(toUser, org.Value, parameters)
+                            .Match(
```

Comment:
You can use Bind here (then you won't have to provide the error)

Reply:
The compiler doesn't like it as it's a maybe, and not a result